src/main/java/de/mediathekview/mserver/crawler/ard/json/ArdMediaArrayToDownloadUrlsConverter.java

```diff
@@ -32,52 +35,152 @@ private ArdMediaArrayToDownloadUrlsConverter() {}
+              relevantUrl = url;
+            }
+            break;
+          default:
+            if (usedFileType.isEmpty()) {
+              usedFileType = fileType.get();
+              relevantUrl = url;
+            }
+        }
+      }
+    }
+    
+    return relevantUrl;
+  }
+  
+  private static void parseMediaArray(JsonArray aMediaArray, Map<Resolution, Set<String>> aUrls) {
+    
+    aMediaArray.forEach(mediaEntry -> {
+      final JsonArray mediaStreamArray = mediaEntry.getAsJsonObject()
+              .getAsJsonArray(ELEMENT_MEDIA_STREAM_ARRAY);
+
+      parseMediaStreamArray(mediaStreamArray, aUrls);
+    });    
+  }
+
+  private static void parseMediaStreamArray(JsonArray aMediaStreamArray, Map<Resolution, Set<String>> aUrls) {
+
+    for (int i = 0; i < aMediaStreamArray.size(); i++) {
+      final JsonElement videoElement = aMediaStreamArray.get(i);
+      final String qualityAsText =
+          videoElement.getAsJsonObject().get(ELEMENT_QUALITY).getAsString();
+      
+      Optional<Resolution> quality = getQuality(qualityAsText);
+      if (quality.isPresent()) {
+        String downloadUrl = "";
+        
+        if (videoElement.getAsJsonObject().has(ELEMENT_SERVER)) {
+          final String baseUrl = videoElement.getAsJsonObject().get(ELEMENT_SERVER).getAsString();
+          downloadUrl = videoElementToUrl(videoElement, baseUrl);
+          addUrl(downloadUrl, quality.get(), aUrls);
+        } 
+        
+        if (downloadUrl.isEmpty() && videoElement.getAsJsonObject().has(ELEMENT_STREAM)) {
+          JsonElement streamObject = videoElement.getAsJsonObject().get(ELEMENT_STREAM);
+          if (streamObject.isJsonPrimitive()) {
+            final String baseUrl = streamObject.getAsString();
+            downloadUrl= videoElementToUrl(videoElement, baseUrl);
+            addUrl(downloadUrl, quality.get(), aUrls);          
+          } else if(streamObject.isJsonArray()) {
+            JsonArray streamArray = streamObject.getAsJsonArray();
+            streamArray.forEach(stream -> {
+              final String baseUrl = stream.getAsString();
+              addUrl(baseUrl, quality.get(), aUrls);          
+            });
+          }
+        }
+      }
+    }
+  }
+  
+  private static void addUrl(final String aUrl, 
+    final Resolution quality, 
+    Map<Resolution, Set<String>> aUrls) {
+
+    if (!aUrl.isEmpty()) {
+      if (aUrl.startsWith(PROTOCOL_RTMP)) {
+        LOG.debug("Found an Sendung with the old RTMP format: " + aUrl);
+      } else {
+        aUrls.get(quality).add(aUrl);
+      }
     }
-    return downloadUrls;
   }
+  
+  private static Optional<Resolution> getQuality(String aQualityAsText) {
+    int qualityNumber;
+    try {
+      qualityNumber = Integer.parseInt(aQualityAsText);
+    } catch (final NumberFormatException numberFormatException) {
+     /* LOG.debug("Can't convert quality %s to an integer.", aQualityAsText,
+          numberFormatException);*/
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)